src/algorithms/tracking/TrackSeeding.cc

```diff
@@ -239,18 +239,18 @@ std::unique_ptr<edm4eic::TrackParametersCollection> eicrecon::TrackSeeding::make
       trackparam.setTime(10); // time in ns
       #if EDM4EIC_VERSION_MAJOR >= 5
         edm4eic::Cov6f cov;
-        cov(0,0) = 0.1; // loc0
-        cov(1,1) = 0.1; // loc1
-        cov(2,2) = 0.05; // theta
-        cov(3,3) = 0.05; // phi
-        cov(4,4) = 0.05; // qOverP
-        cov(5,5) = 0.1; // time
+        cov(0,0) = m_cfg.m_Loc_a_Error; // loc0
+        cov(1,1) = m_cfg.m_Loc_b_Error; // loc1
+        cov(2,2) = m_cfg.m_theta_Error; // theta
+        cov(3,3) = m_cfg.m_phi_Error; // phi
```

Comment:
We should use the same ordering as in Acts here: https://github.com/acts-project/acts/blob/3eb0b4febd16ad8cee5664302e3542044ab17d35/Core/include/Acts/Definitions/TrackParametrization.hpp#L37-L56

```suggestion
        cov(2,2) = m_cfg.m_phi_Error; // phi
        cov(3,3) = m_cfg.m_theta_Error; // theta
```

And you may want to double check that this is then read in correctly where the seeds are used too.

Reply:
I guess this should be adjusted for the truth-seeding as well then:

[https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/TrackParamTruthInit.cc#L139-L140](https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/TrackParamTruthInit.cc#L139-L140)

Reply:
Based on this code in CKFTracking.cc:

```
Acts::BoundSquareMatrix cov = Acts::BoundSquareMatrix::Zero();
#if EDM4EIC_VERSION_MAJOR >= 5
  for (size_t i = 0; const auto& [a, x] : edm4eic_indexed_units) {
   for (size_t j = 0; const auto& [b, y] : edm4eic_indexed_units) {
      cov(a, b) = track_parameter.getCovariance()(i,j) * x * y;
      ++j;
   }
   ++i;
  }
```

I think that, for example, the `cov(2,2)` value set in TrackSeeding.cc will be treated as initial value of `cov(2,2)` in CKFTracking.cc. Is that correct @wdconinc?

But this means that Jeet's studies of the impact of the initial theta (phi) error might have actually been studies of the initial phi (theta) error. So, we should probably check this next week with the correct indexing.

Reply:
Sure. I can do that. 

But why do you think something needs to be changed in CKFTracking? It looks like it is reading the correct indices for the initial covariance matrix. It is just that cov(2,2) corresponds to phi and cov(3,3) corresponds to theta.

So, in the truth seeding code (lines 139-140), where we currently do have different initial errors for theta and phi, we should change

 ```
cov(2,2) = 0.01; // theta
cov(3,3) = 0.05; // phi
```

to

 ```
cov(2,2) = 0.05; // phi
cov(3,3) = 0.01; // theta
```

And that would correct it (for the truth-seeding case)?

Reply:
I guess you mean these lines should change:

[https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/CKFTracking.cc#L72-L73](https://github.com/eic/EICrecon/blob/main/src/algorithms/tracking/CKFTracking.cc#L72-L73)

Reply:
I think I'll retract my previous statement that we might have been previously been studying the effect of the initial theta (phi) when we thought we were looking at the initial phi (theta) error. If the loop was reading in from `edm4eic_indexed_units` in order, then it was previously doing something like:

 `cov(Acts::eBoundTheta, Acts::eBoundTheta) =  track_parameter.getCovariance()(2,2);`

with `Acts::eBoundTheta==3`.  So, previously the CKF was being passed the correct error for theta and phi, although in a confusing way.

If that is the case, there should not be any effect from this new indexing. But, @Jeet-bhai and I can double check next week.

Reply:
@bschmookler I'm ok with resolving this conversation (i.e. no outstanding concerns). Are you too? If so, please click resolve.

Then we can get back to overall review and approval.